tests/unit/test_refresh_lock_registry.py

```
@@ -0,0 +1,254 @@
+    """Reset module state between tests."""
+    auth_mod._REFRESH_GENERATIONS.clear()
+    # The WeakKeyDictionary is mutated indirectly; we don't reset it here so
+    # the cleanup test can observe natural GC behavior.
+    yield
+    auth_mod._REFRESH_GENERATIONS.clear()
+
+
+class TestPerLoopLockIdentity:
+    def test_different_loops_get_different_locks(self):
+        """A lock created in loop X must NOT be returned to loop Y."""
+        path = Path("/tmp/notebooklm-test/storage_state.json")
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/notebooklm-test/storage_state.json"

(S108)